### PR TITLE
[skip-tests] Add a preset that builds all configs of all projects.

### LINKED ISSUE #579: [FEA]: Add CMake preset for development that enables all projects
### Is this a duplicate?

- [X] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this request and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Area

Infrastructure

### Is your feature request related to a problem? Please describe.

Current presets only enable a single project. Changing presets between projects is annoying and gets in the way of developing features that span subprojects.

### Describe the solution you'd like

Add a preset that:

- Enables all projects
- Enables all multiconfig options for Thrust/CUB
- Turns on `CUB_SEPARATE_CATCH2` to reduce cost of iterative build/test cycles on CUB catch tests.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by allisonvacanti**
## Description

closes #579.

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -226,6 +226,37 @@
...
+        "THRUST_MULTICONFIG_WORKLOAD": "LARGE",
+        "THRUST_IGNORE_DEPRECATED_CPP_DIALECT": true,
+        "THRUST_MULTICONFIG_ENABLE_DIALECT_CPP11": true,
+        "THRUST_MULTICONFIG_ENABLE_DIALECT_CPP14": true,
+        "THRUST_MULTICONFIG_ENABLE_DIALECT_CPP17": true,
+        "THRUST_MULTICONFIG_ENABLE_DIALECT_CPP20": true,
+        "THRUST_MULTICONFIG_ENABLE_SYSTEM_CPP": true,
+        "THRUST_MULTICONFIG_ENABLE_SYSTEM_CUDA": true,
+        "THRUST_MULTICONFIG_ENABLE_SYSTEM_OMP": true,
+        "THRUST_MULTICONFIG_ENABLE_SYSTEM_TBB": true
```
Comment (miscco) [inline CMakePresets.json:258]: 👀

**Review by miscco — APPROVED** [review]

File: CMakePresets.json (hunk 1 above)
Comment (allisonvacanti) [reply CMakePresets.json:258]: Thrust's build system is a *beast*: https://github.com/NVIDIA/cccl/blob/main/thrust/cmake/ThrustMultiConfig.cmake#L40-L70 😄

**Comment by allisonvacanti** [comment]
Just added skip-tests in forced push. This doesn't change anything that builds in CI.

File: CMakePresets.json [hunk 2]
```diff
@@ -226,6 +226,37 @@
...
+    },
+    {
+      "name": "all-dev",
+      "inherits": "base",
+      "cacheVariables": {
+        "CCCL_ENABLE_LIBCUDACXX": true,
+        "CCCL_ENABLE_CUB": true,
+        "CCCL_ENABLE_THRUST": true,
+        "CCCL_ENABLE_TESTING": true,
+        "CCCL_ENABLE_EXAMPLES": true,
```
Comment (senior-zero) [inline CMakePresets.json:238]: suggestion: + CCCL_ENABLE_BENCHMARKS

**Review by senior-zero — APPROVED** [review]

**Pull request merged**